pkg/k8shandler/logstore.go

```diff
@@ -6,18 +6,43 @@ import (
 	"k8s.io/apimachinery/pkg/api/errors"
 
 	"github.com/openshift/cluster-logging-operator/pkg/utils"
+	"github.com/sirupsen/logrus"
+	"reflect"
 
 	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
 	sdk "github.com/operator-framework/operator-sdk/pkg/sdk"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func CreateOrUpdateLogStore(logging *logging.ClusterLogging) (err error) {
-	if err = createOrUpdateElasticsearchSecret(logging); err != nil {
-		return
+
+	if logging.Spec.LogStore.Type == "elasticsearch" {
```

Comment:
Can you source "elasticsearch" from a common conststants file?

Reply:
Enum might be sufficient for this
```go
const (
  Elasticsearch = 0
  Fluentd = 1
  // ...
)
```

Reply:
Addressed by creating additional types